datafusion/src/scalar.rs

```diff
@@ -1901,5 +2037,191 @@ mod tests {
             )),
             None
         );
+
```

Comment:
These are very nice and thorough tests.

The only thing I didn't see a test for was struct recursion -- like

```
A : {
  B: "foo"
  C: "bar"
}
```

I wonder if you could extend the `A`, `B`, `C` example below to include a nested struct as well?

Reply:
Good idea. Done in 62ad373